presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java

```diff
@@ -152,23 +160,30 @@ public DockerContainer withHealthCheck(Path healthCheckScript)
                 .withCreateContainerCmdModifier(command -> command.withHealthcheck(cmd));
     }
 
-    public Duration getStartupTime()
+    public synchronized Duration getStartupTime()
     {
-        return Duration.succinctNanos(startupTime.elapsed(NANOSECONDS)).convertToMostSuccinctTimeUnit();
+        checkState(lastStartUpCommenceTimeNanos.isPresent(), "Container did not commence starting");
+        checkState(lastStartUFinishTimeNanos.isPresent(), "Container not started");
+        return Duration.succinctNanos(lastStartUFinishTimeNanos.getAsLong() - lastStartUpCommenceTimeNanos.getAsLong()).convertToMostSuccinctTimeUnit();
     }
 
     @Override
     protected void containerIsStarting(InspectContainerResponse containerInfo)
     {
-        this.startupTime.start();
+        synchronized (this) {
+            lastStartUpCommenceTimeNanos = OptionalLong.of(System.nanoTime());
```

Comment:
this should also reset `lastStartUFinishTimeNanos`

Reply:
good catch, fixed

Reply:
why?
It shoud verify if `lastStartUFinishTimeNanos` is empty. Rigth?

Reply:
The whole point is that starts are retried (as per cmt msg  and @kokosing analysis in https://github.com/prestosql/presto/issues/6079)

Reply:
Ok - I thought that they are required only if thy do not reach actual "started" state.